packages/graphql_server/src/ScrapeRepos.ts

```diff
@@ -0,0 +1,231 @@
+    const repoName = html(el).text().trim()
+    repos.push(repoName)
+  })
+
+  const trendingSplit: string[] = []
+  // trim the repos to be correctly formatted
+  repos.forEach((repo) => {
+    const trimmedName = repo.replace(/\n\s+/g, '').replace(/\//g, '')
+    const stringSplit = trimmedName.split(' ')
+    trendingSplit.push(stringSplit[0])
+    trendingSplit.push(stringSplit[1])
+  })
+  return trendingSplit
+}
+
+/**  This function imports the ReadMe.md file for a repository (if it can be located)
+ * @param {string} owner - owner of the repo
+ * @param {string} name - name of the repo
+ * @returns {string} a string containing the text of the repo. Throws an error if file can't be located
+ * @todo paths in the beginning can be constantly adapted if new ReadMe file locations are being found
+ */
+async function getReadme(owner: string, name: string) {
+  // these paths exists to check in multiple locations for the readme files
+  const readmePaths: string[] = [
+    `https://raw.githubusercontent.com/${owner}/${name}/release/readme.md`,
+    `https://raw.githubusercontent.com/${owner}/${name}/dev/README.rst`,
+    `https://raw.githubusercontent.com/${owner}/${name}/main/README.md`,
+    `https://raw.githubusercontent.com/${owner}/${name}/master/README.md`
+  ]
+
+  for (let i = 0; i < readmePaths.length; i++) {
+    try {
+      const response: AxiosResponse<string> = await axios.get(readmePaths[i])
+      const converter = new showdown.Converter()
+
+      // Use the converter object to convert Markdown to HTML to String:
+      const html: string = converter.makeHtml(response.data)
+      const html2: string = html.toString()
+      return html2
+        .replace(/<[^>]*>/g, '')
+        .replace(/\n\s+/g, '')
+        .replace(/\//g, '')
+    } catch (error) {
+      continue
+    }
+  }
+  throw new Error("ReadMe couldn't be found")
+}
+
+/** Gets the repo's information via GitHub's GraphQL API
+ * @param {string} query GraphQL query for the repo (including owner and name)
+ * @param {string} authToken personal authorization token
+ * @returns {any[]} the json data for the requested repo as by the graphql query; null on error
+ */
+async function getRepoInfo(query: string, authToken: string): Promise<Repository | null> {
+  try {
+    const response: AxiosResponse<{ data: { repository: Repository } }> = await axios.post(
+      'https://api.github.com/graphql',
+      {
+        query: query
+      },
+      {
+        headers: {
+          Authorization: authToken
+        }
+      }
+    )
+    return response.data.data.repository
+  } catch (error) {
+    return null
+  }
+}
+
+/** Creates the ELI5 of the repository by passing the readme to chatgpt
+ * @param {string} readME readMe file of the repository as a string
+ * @param {string} OPENAI_API_KEY api key for OpenAI
+ * @returns {string} the summary by ChatGPT of the repo or on error null
+ */
+async function openAIRequestTurbo(readME: string, OPENAI_API_KEY: string) {
```

Comment:
place in file `eli5.ts`, rename to `getELI5DescriptionForRepositoryFromText`, or suggest ideas about how to structure api.
Why this change? Explaination: scraping repositorys has to do nothing with providing a description about repositories, the files should be named accordingly to what they are supposed to do.

Reply:
the function will be reused outside of the trending repo context